src/internal/clusterstate/v2.6.0/pfsdb.go

```diff
@@ -460,6 +460,10 @@ func traverseToEdges(startCommit *v2_5_0.CommitInfo, skipSet map[string]*v2_5_0.
 //
 // 3) commit keys can now be substituted from <project>/<repo>@<branch>=<id> -> <project>/<repo>@<id>
 func branchlessCommitsPFS(ctx context.Context, tx *pachsql.Tx) error {
+	// enforce known DB invariants
+	if err := deleteDanglingCommitRefs(ctx, tx); err != nil {
```

Comment:
nit, I think this should be called in the apply callback rather than in the core migration logic.